registries/_format/base64url.md

```diff
@@ -16,6 +16,107 @@ Base type: `{{ page.base_type }}`.
 
 The `{{page.slug}}` format is a url-safe binary array as defined in [RFC4648](https://www.rfc-editor.org/rfc/rfc4648#section-5).
 
+### Upgrading from < 3.1 to >= 3.1
+
+#### Entire response body
+
+```yaml
+#OAS 3.0
+paths:
+  '/picture':
+    get:
+      responses:
+        '200':
+          content:
+            image/png:
+              schema:
+                type: string
+                format: byte
```

Comment:
I think this example doesn't make sense. `"content":{"image/png":...}` means that the response will have `Content-Type: image/png`, and the response body should be the raw octet stream of a PNG image.

A correct example would be

~~~yaml
#OAS 3.0
paths:
  '/picture':
    get:
      responses:
        '200':
          content:
            image/png:
              schema:
                type: string
                format: binary
~~~

And it would belong to file `binary.md`.

A "correct" entire response body example for `format: base64url` would be with `Content-Type: text/plain`:

~~~yaml
#OAS 3.0
paths:
  '/picture':
    get:
      responses:
        '200':
          content:
            text/plain:
              schema:
                type: string
                format: base64url
~~~

However I am at a loss how the API would tell the client what is inside the base64url text, that would probably require out-of-band information, for example in a `description` field.

Reply:
IIRC, when we were working through this for OAS 3.1, @darrelmiller said that in this case `Content-Type` would still be `image/png` because it is about the unencoded payload.  I might be misremembering _who_ argued for that, but it was someone because I remember having to change it 

The `contentEncoding` variation on this example is the [exact example](https://spec.openapis.org/oas/v3.1.0#considerations-for-file-uploads) in the OAS 3.1 spec, btw.  It comes with the following paragraph:

> Note that the `Content-Type` remains `image/png`, describing the semantics of the payload. The JSON Schema `type` and `contentEncoding` fields explain that the payload is transferred as text. The JSON Schema `contentMediaType` is technically redundant, but can be used by JSON Schema tools that may not be aware of the OpenAPI context.

I agree that it is a very confusing case, and as noted elsewhere in this discussion, there's not much reason to use it at all anyway.

The [corresponding example](https://spec.openapis.org/oas/v3.0.3#considerations-for-file-uploads) in OAS 3.0 is more ambiguous (note that that spec is confused over `byte` vs `base64` for the correct value to use with `format`).

Note that there was never a way to indicate the encoded media type in OAS 3.0 unless it was in the key for the Media Type Object or a scenario where an Encoding Object was used.

Reply:
Should we merge that as is and revisit when https://github.com/OAI/OpenAPI-Specification/issues/3174 is adressed?

Reply:
I kind of feel like we should omit the examples that don't make sense (even if they are currently present in the spec).  Let me think a bit- I'm going to write a summary comment about finishing this PR shortly.